bin/before_install

```diff
@@ -4,6 +4,39 @@ if [ -n "$CI" ]; then
   echo "== Installing system packages =="
   sudo apt-get install libcurl4-openssl-dev
   echo
+
+  # Install Qpid proton system library
+  set -v
+  pushd $PWD
+
+    # Install the dev dependencies for building Qpid proton system library.
+    sudo apt-get install -y gcc cmake cmake-curses-gui uuid-dev
+    sudo apt-get install -y libssl-dev
+    sudo apt-get install -y libsasl2-2 libsasl2-dev
+
+    # Get the latest Qpid Proton source
+    cd $HOME/build
+    git clone --branch 0.30.0 https://github.com/apache/qpid-proton.git
+    cd qpid-proton
+
+    # Configure the source of Qpid Proton.
+    mkdir build
+    cd build
+    cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_BINDINGS=
+
+    # Compile system libraries.
+    make all
+
+    # Install system libraries
+    sudo make install
+
+    # Enable the qpid_proton bundler group
+    [ -z "$BUNDLE_WITH" ] && bundle config with qpid_proton
```

Comment:
@agrare Technically this failed with 

```
bin/before_install: line 34: bundle: command not found
```

because we don't have Ruby yet, and I'm not sure why it doesn't have BUNDLE_WITH in the env like in https://github.com/ManageIQ/manageiq-providers-nuage/pull/267

Reply:
Does it work with BUNDLE_WITH like nuage has?

Reply:
It seems to work regardless (both repos passed), but maybe the tests themselves are gated somehow that I'm not seeing?

Reply:
We do `config.filter_run_excluding(:qpid_proton) unless ENV['CI'] || Gem.loaded_specs.key?(:qpid_proton)` so it looks like this would exclude those tests if qpid_proton weren't installed.

Seems like we'd want to never skip on `ENV['CI']` regardless of if qpid_proton were available, since that would indicate something was wrong with the env setup.

Okay I'll convert this to use the env var like nuage and make sure the qpid_proton event_catcher specs are running.